File: Graph/GFG/Easy/AdjacencyListDFS/AdjacencyListDFS.java
// https://www.geeksforgeeks.org/problems/depth-first-traversal-for-a-graph/1
import java.util.ArrayList;
public class AdjacencyListDFS {
    public ArrayList<Integer> dfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj) {
        // Code here
        ArrayList<Integer> dfs = new ArrayList<>();
        int visited[] = new int[V];
        DFS(0, adj, visited, dfs);
        return dfs;
    }

    void DFS(int start, ArrayList<ArrayList<Integer>> adj, int visited[], ArrayList<Integer> dfs) {
        visited[start] = 1;
        dfs.add(start);
        for (int ele : adj.get(start)) {
            if (visited[ele] == 0) {
                DFS(ele, adj, visited, dfs);
            }
        }
    }
}

File: Graph/GFG/Easy/AdjacencyListDFS/AdjacencyListDFS.cpp
// https://www.geeksforgeeks.org/problems/depth-first-traversal-for-a-graph/1
#include<bits/stdc++.h>
using namespace std;

void DFS(int start, vector<int> adj[], vector<int> &dfs, vector<int> &visited)
{
    visited[start] = 1;
    dfs.push_back(start);
    for (int neighbor : adj[start])
    {
        if (visited[neighbor] == 0)
        {
            DFS(neighbor, adj, dfs, visited);
        }
    }
}

vector<int> dfsOfGraph(int V, vector<int> adj[])
{
    // Code here
    vector<int> dfs;
    vector<int> visited(V, 0);
    DFS(0, adj, dfs, visited);
    return dfs;
}
int main(void) {
    
    return 0;
}